src/main/java/it/gov/pagopa/pu/debtpositions/service/update/DebtPositionManageInstallmentsServiceImpl.java

```diff
@@ -0,0 +1,131 @@
+      throw new ConflictErrorException(String.format("Payment option having id %s cannot be modified because is not in allowed status: %s", storedPaymentOption.getPaymentOptionId(), storedPaymentOption.getStatus()));
+    }
+
+    storedPaymentOption.setDescription(manageDebtPositionDTO.getPaymentOptionDescription());
+
+    List<InstallmentDTO> installment2operate = managePaymentOptionInstallments(storedDebtPosition, storedPaymentOption, manageDebtPositionDTO.getInstallments(), accessToken, operatorExternalUserId);
+
+    Pair<DebtPositionDTO, String> debtPositionManaged = execute(storedDebtPosition, installment2operate, wfExecutionParameters, PaymentEventType.DP_UPDATED, accessToken, operatorExternalUserId);
+
+    log.debug("Managed installments for debt position with id {}", storedDebtPosition.getDebtPositionId());
+    return debtPositionManaged;
+  }
+
+  private List<InstallmentDTO> managePaymentOptionInstallments(DebtPositionDTO debtPositionDTO, PaymentOptionDTO paymentOptionDTO, List<ManageInstallmentDTO> manageInstallments, String accessToken, String operatorExternalUserId) {
+    ArrayList<InstallmentDTO> installmentsToAdd = new ArrayList<>();
+    ArrayList<InstallmentDTO> installmentsToUpdate = new ArrayList<>();
+    ArrayList<InstallmentDTO> installmentsToCancel = new ArrayList<>();
+
+    manageInstallments.forEach(
+      manageInstallmentDTO -> {
+        InstallmentDTO manageInstallment = manageInstallmentDTO.getInstallment();
+        switch (manageInstallmentDTO.getAction()) {
+          case I -> {
+            paymentOptionDTO.addInstallmentsItem(manageInstallment);
+            installmentsToAdd.add(manageInstallment);
+          }
+          case M -> {
+            InstallmentDTO storedInstallment = findInstallmentToManage(paymentOptionDTO, manageInstallment.getInstallmentId());
+            debtPositionManageApplierService.merge(manageInstallment, storedInstallment);
+            installmentsToUpdate.add(storedInstallment);
+          }
+          case A -> {
+            InstallmentDTO storedInstallment = findInstallmentToManage(paymentOptionDTO, manageInstallment.getInstallmentId());
+            installmentsToCancel.add(storedInstallment);
+          }
+        }
+      }
+    );
+
+    WfExecutionParameters wfExecutionParameters = WfExecutionParameters.builder().massive(false).partialChange(true).build();
+
+    DebtPositionDTO dpWithInstallmentsAdded = debtPositionAddInstallmentService.addInstallment(debtPositionDTO, installmentsToAdd, wfExecutionParameters, accessToken, operatorExternalUserId).getLeft();
+    DebtPositionDTO dpWithInstallmentsUpdated = debtPositionUpdateInstallmentService.updateInstallment(dpWithInstallmentsAdded, installmentsToUpdate, wfExecutionParameters, accessToken, operatorExternalUserId).getLeft();
+    debtPositionCancelInstallmentService.cancelInstallment(dpWithInstallmentsUpdated, installmentsToCancel, wfExecutionParameters, accessToken, operatorExternalUserId);
```

Comment:
before to start a new operation, you should check if the method returned a not null workflowId
if not null, you should call the workflow status API until a terminal status has been returned.

You could copy the Service from activities:
it.gov.pagopa.payhub.activities.service.ingestionflow.debtposition.DPInstallmentsWorkflowCompletionService

Or, we could expose an API on workflow-hub to wait completion (on workflow-hub before to use the Service, you need to import the Service on the main Class)